packages/cli/src/LoadNodesAndCredentials.ts

```diff
@@ -91,48 +81,10 @@ export class LoadNodesAndCredentials implements INodesAndCredentials {
 
 		await this.loadNodesFromCustomDirectories();
 		await this.postProcessLoaders();
-		this.injectCustomApiCallOptions();
 	}
 
-	async generateTypesForFrontend() {
-		const credentialsOverwrites = CredentialsOverwrites().getAll();
-		for (const credential of this.types.credentials) {
-			const overwrittenProperties = [];
-			this.credentialTypes
-				.getParentTypes(credential.name)
-				.reverse()
-				.map((name) => credentialsOverwrites[name])
-				.forEach((overwrite) => {
-					if (overwrite) overwrittenProperties.push(...Object.keys(overwrite));
-				});
-
-			if (credential.name in credentialsOverwrites) {
-				overwrittenProperties.push(...Object.keys(credentialsOverwrites[credential.name]));
-			}
-
-			if (overwrittenProperties.length) {
-				credential.__overwrittenProperties = uniq(overwrittenProperties);
-			}
-		}
-
-		// pre-render all the node and credential types as static json files
-		await mkdir(path.join(GENERATED_STATIC_DIR, 'types'), { recursive: true });
-
-		const writeStaticJSON = async (name: string, data: object[]) => {
-			const filePath = path.join(GENERATED_STATIC_DIR, `types/${name}.json`);
-			const stream = createWriteStream(filePath, 'utf-8');
-			stream.write('[\n');
-			data.forEach((entry, index) => {
-				stream.write(JSON.stringify(entry));
-				if (index !== data.length - 1) stream.write(',');
-				stream.write('\n');
-			});
-			stream.write(']\n');
-			stream.end();
-		};
-
-		await writeStaticJSON('nodes', this.types.nodes);
-		await writeStaticJSON('credentials', this.types.credentials);
+	addPostProcessor(fn: () => Promise<void>) {
```

Comment:
I take it we went for this to avoid adding more dependencies.

There are only two post-processors for now, but over time I worry it can become rather hard to keep track of what's getting added and later called when and in what order.

Reply:
yeah. I couldn't come up with a better way to decouple this processing pipeline. This works for now, but I'm very much open to suggestions.

Reply:
I'll keep thinking if there's another way but let's not block on this.